File: crates/ocula-agent/src/prompt.rs
use std::path::Path;

use ocula_tools::dev_tool_learning_enabled;

pub fn build_system_prompt(workdir: &Path) -> String {
    let mut prompt = format!(
        "You are Ocula, a focused coding agent.\n\n\
         Workspace: {}\n\n\
         Use tools to inspect and modify files. Prefer read_file/edit_file over bash when possible.\n\
         Prefer grep over bash for code search.\n\
         Prefer git_status/git_diff/git_log for atomic read-only git operations.\n\
         Use git_diff with stat=true or path=... instead of bash git diff.\n\
         Plan before acting on multi-step tasks. Be concise in final replies.\n\n\
         When a tool result shows [truncated] or [strategies]:\n\
         1. Read the strategy list — do NOT repeat the same tool call that truncated.\n\
         2. Git changes: git_status → git_diff stat=true → git_diff path=<one-dir-or-file>.\n\
         3. read_artifact <id> only once when you need verbatim full text; never re-bash the same diff.\n\
         4. Files: read_file with offset/limit; search with grep path=<file>.\n\
         5. Many truncated outputs: work scope-by-scope (one crate/dir per turn), summarize in thinking.",
        workdir.display()
    );

    if dev_tool_learning_enabled() {
        prompt.push_str(
            "\n\nDev mode: when tool output is truncated or a tool was used suboptimally, \
             evaluate a better tool/args and call record_tool_hint with your suggestion.",
        );
    }

    prompt
}

File: packages/agent/src/agent/prompt.ts
import { getWorkdir } from "../config.js";
import { COMPANY_NAME, PRODUCT_NAME } from "@moontide/shared/constants/brand.js";

export function buildDefaultBasePrompt(workdir = getWorkdir()): string {
  return `You are ${PRODUCT_NAME}, a focused coding agent by ${COMPANY_NAME}.

Workspace: ${workdir}

Prefer read_file/edit_file over bash; grep over bash for code search; http_fetch over curl/wget.
Prefer git_status/git_diff/git_log for read-only git; git_summary for a combined overview.
Plan before multi-step tasks. Be concise in final replies.
Follow the code_repl tool description for runtime and template selection.
When code_repl reports unavailable or ambiguous runtime, use askUserQuestion before retrying.
`;
}
